test/e2e_tests/sync.go

```diff
@@ -15,10 +15,37 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const NamespaceManifestWorkPrefix = "mw-create-"
+const (
+	namespaceManifestWorkPrefix = "mw-create-"
+	envVarName                  = "E2E-TEST"
+)
+
+// verifySecretOrConfigMapCopy makes sure a Secret or a ConfigMap is eventually copied to the ManifestWork,
+// by creating a Capp with the needed configuration and getting the created ManifestWork resource.
+func verifySecretOrConfigMapCopy(baseCapp *cappv1alpha1.Capp, name, namespace, kind string) {
+	desiredCapp := utilst.CreateCapp(k8sClient, baseCapp)
+
+	assertionCapp := utilst.GetCappWithPlacementAnnotation(k8sClient, desiredCapp.Name, desiredCapp.Namespace)
+	mwNamespace := assertionCapp.Annotations[testconsts.AnnotationKeyHasPlacement]
+
+	manifestWork := &workv1.ManifestWork{}
+
+	resourceObject := map[string]client.Object{
```

Comment:
If you change it to factory and not init another car call it resourceFactory. Cause this map signifies mini factory design